controllers/core/openstackcontrolplane_controller.go

```diff
@@ -559,6 +579,17 @@ func (r *OpenStackControlPlaneReconciler) SetupWithManager(
 	}); err != nil {
 		return err
 	}
+	// index topologyField
+	if err := mgr.GetFieldIndexer().IndexField(ctx, &corev1beta1.OpenStackControlPlane{}, topologyField, func(rawObj client.Object) []string {
+		// Extract the topology name from the spec, if one is provided
+		cr := rawObj.(*corev1beta1.OpenStackControlPlane)
+		if cr.Spec.TopologyRef == nil {
+			return nil
+		}
+		return []string{cr.Spec.TopologyRef.Name}
+	}); err != nil {
+		return err
+	}
```

Comment:
Do we have to index it in the ctlplane controller? If I have not missed something, we are not using the content of the topology here, so the only situation we care is the change of the name and thats a change we get also without indexing the field.

Reply:
Right, we do not need this because it's just about passing parameters to the underlying resources. I removed it, and I removed the Topology watch as well: for any content change we do not need to trigger a reconciliation here, but we trigger the reconciliation on the services that are already watching this topology.

Reply:
(Done, resolving the thread unless there's something else to address)